ci/create-windows-image.ps1

```diff
@@ -0,0 +1,274 @@
+    param(
+        [Parameter(Mandatory=$true, Position=0)]
+        [string] $ZipFile,
+        [Parameter(Mandatory=$true, Position=1)]
+        [string] $OutPath
+    )
+
+    [System.IO.Compression.ZipFile]::ExtractToDirectory($zipFile, $outPath)
+}
+
+# Set various known paths
+$openSSHZip = Join-Path $env:TEMP 'OpenSSH.zip'
+$openSSHInstallDir = Join-Path $env:ProgramFiles 'OpenSSH'
+$openSSHInstallScript = Join-Path $openSSHInstallDir 'install-sshd.ps1'
+$openSSHDownloadKeyScript = Join-Path $openSSHInstallDir 'download-key-pair.ps1'
+$openSSHDaemon = Join-Path $openSSHInstallDir 'sshd.exe'
+$openSSHDaemonConfig = [io.path]::combine($env:ProgramData, 'ssh', 'sshd_config')
+
+# Download and unpack the binary distribution of OpenSSH
+Invoke-WebRequest -Uri $openSSHURL `
+    -OutFile $openSSHZip `
+    -ErrorAction Stop
+
+Unzip -ZipFile $openSSHZip `
+    -OutPath "$env:TEMP" `
+    -ErrorAction Stop
+
+Remove-Item $openSSHZip `
+    -ErrorAction SilentlyContinue
+
+# Move into Program Files
+Move-Item -Path (Join-Path $env:TEMP 'OpenSSH-Win64') `
+    -Destination $openSSHInstallDir `
+    -ErrorAction Stop
+
+# Run the install script, terminate if it fails
+& Powershell.exe -ExecutionPolicy Bypass -File $openSSHInstallScript
+if ($LASTEXITCODE -ne 0) {
+	throw("Failed to install OpenSSH Server")
+}
+
+# BEGIN IMPROBABLE - Allow read and execute for Users
+$acl = Get-ACL -Path $openSSHInstallDir
+$ar = New-Object System.Security.AccessControl.FileSystemAccessRule( `
+	"BUILTIN\Users", "ReadAndExecute", "Allow")
+$acl.AddAccessRule($ar)
+Set-Acl -Path $openSSHInstallDir -AclObject $acl
+# END IMPROBABLE
+
+# Add a firewall rule to allow inbound SSH connections to sshd.exe
+New-NetFirewallRule -Name sshd `
+    -DisplayName "OpenSSH Server (sshd)" `
+    -Group "Remote Access" `
+    -Description "Allow access via TCP port 22 to the OpenSSH Daemon" `
+    -Enabled True `
+    -Direction Inbound `
+    -Protocol TCP `
+    -LocalPort 22 `
+    -Program "$openSSHDaemon" `
+    -Action Allow `
+    -ErrorAction Stop
+
+# Ensure sshd automatically starts on boot
+Set-Service sshd -StartupType Automatic `
+    -ErrorAction Stop
+
+# Set the default login shell for SSH connections to Powershell
+New-Item -Path HKLM:\SOFTWARE\OpenSSH -Force
+New-ItemProperty -Path HKLM:\SOFTWARE\OpenSSH `
+    -Name DefaultShell `
+    -Value "C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe" `
+    -ErrorAction Stop
+
+$keyDownloadScript = @'
+# Download the instance key pair and authorize Administrator logins using it
+$openSSHAdminUser = 'c:\ProgramData\ssh'
+$openSSHAuthorizedKeys = Join-Path $openSSHAdminUser 'authorized_keys'
+
+If (-Not (Test-Path $openSSHAdminUser)) {
+    New-Item -Path $openSSHAdminUser -Type Directory
+}
+
+# BEGIN IMPROBABLE - use Google services
+# See https://cloud.google.com/compute/docs/storing-retrieving-metadata#default
+$keyUrl = "http://metadata.google.internal/computeMetadata/v1/project/attributes/ssh-keys"
```

Comment:
This will work only if run in a GCE project.  Is that a problem for this config, since it's hosted in the public repo?

I guess, to rephrase, since this will be visible to people who don't have access to our infrastructure, is it OK that this will not work for them?

Reply:
This is meant as a guide for setting up Ansible in the platform repo later.  I'll add a comment to the effect that "This is our stuff, you can use it as a guide, but it's not useful to you for more than that."